service/src/internal/runner.rs

```diff
@@ -247,30 +249,22 @@ impl PdaRunner {
                         }
                         Err(e) => {
                             error!("{job:?} failed to request a proof: {e}");
-                            job_status = JobStatus::Failed(
-                                e,
-                                Some(JobStatus::RemoteZkProofRequesting.into()),
-                            );
-                            self.finalize_job(&job_key, job_status)?;
+                            // NOTE: we internally finalize the job in `handle_zk_client_error`
```

Comment:
This was the root cause, that was missed by AI. even though the diff is in the referenced other PR.